content/courses/connecting-to-offchain-data/oracles.md

```diff
@@ -744,7 +743,7 @@ mod burry_oracle_program {
 
     use super::*;
 
-    pub fn deposit(ctx: Context<Deposit>, escrow_amt: u64, unlock_price: u64) -> Result<()> {
+    pub fn deposit(ctx: Context<Deposit>, escrow_amt: u64, unlock_price: f64) -> Result<()> {
```

Comment:
Why a float? This seems very odd - computers rarely use floats for financial operations, rather using ints for minor units, due to decimal math issues.

Reply:
@mikemaccana  need suggestion : will it be fine to use u64  for unlock_price ?

Reply:
It certainly seems safer - 

 - unsigned (nothing has a negative price)
 - whole numbers rather than decimals
 
 Where did the idea to use f64 come from?

Reply:
(Just a note that I wrote a comment about renaming withdraw_handler, but that's how Anchor does it, so i deleted the comment)

Reply:
i think f64 is used as sol price will be a floating point ex $21.53

Reply:
SOL price will be *shown* as a floating point, but *everything* on Solana is done in whole amounts. Eg, transferring SOL is done in lamports. transferring USDC is done in 2 ^ 6 minor USDC units.

Reply:
This is the same in TradFI too BTW - prices are in cents, pence, etc. although shown in dollars / GBP etc.